code/firmware/rosco_m68k_firmware/duart.asm

```diff
@@ -100,6 +121,34 @@ INITDUART::
     move.l  #EFP_DUMMY_NOOP,EFP_SETCURSOR
     move.l  #CHECKCHAR_DUART,EFP_CHECKCHAR
 
+    ; Setup device blocks...
+    lea.l   DEVICE_BLOCKS,A1
+    move.w  DEVICE_COUNT,D0
+    lsl.w   D0
```

Comment:
Should this be:
```suggestion
    lsl.w   #5,D0
```
to account for the 0x20 size of the device blocks? I think this currently will write starting in the middle of an existing entry is any already exist.

Reply:
Haha good spot, yes it should :D